### PR TITLE
Add toDictionary() to most primitives

### PR DESCRIPTION
**Pull request by f3dm76**

### CODE REVIEW

File: MacawTests/MacawNewSVGTests.swift [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import XCTest
+@testable import Macaw
+
+class MacawNewSVGTests: XCTestCase {
```
Comment (ystrot) [inline MacawTests/MacawNewSVGTests.swift]: Let's remove `New` from the class name.

File: MacawTests/MacawNewSVGTests.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            let node = try SVGParser.parse(bundle: bundle, path: test)
+            guard let serializableNode = node as? Serializable else {
+                XCTFail()
+                return
+            }
+            let jsonData = try JSONSerialization.data(withJSONObject: serializableNode.toDictionary(), options: .prettyPrinted)
+            try jsonData.write(to: URL(fileURLWithPath: path!))
+        } catch {
+            print(error)
+            XCTFail()
```
Comment (ystrot) [inline MacawTests/MacawNewSVGTests.swift]: Why do you print error instead of `XCTFail(error)`?

File: MacawTests/SceneSerialization.swift [hunk 3]
```diff
@@ -0,0 +1,529 @@
...
+        
+        if nodeType == "Shape" {
+            return Shape(dictionary: dictionary)
+        }
+        
+        if nodeType == "Text" {
+            return Text(dictionary: dictionary)
+        }
+        
+        if nodeType == "Group" {
```
Comment (ystrot) [inline MacawTests/SceneSerialization.swift]: Actually all these `if` conditions can be replaced if you will create map of factories: `[<nodeType>:(<dictionary>) -> <node>]`. In this case you don't need to add `init` to each class, but declare `factories` and then call `factories[nodeType]!(dictionary)`

File: MacawTests/SceneSerialization.swift [hunk 4]
```diff
@@ -0,0 +1,529 @@
...
+            return Polyline(dictionary: dictionary)
+        }
+        if type == "Rect" {
+            return Rect(dictionary: dictionary)
+        }
+        if type == "RoundRect" {
+            return RoundRect(dictionary: dictionary)
+        }
+        
+        print("Locus from dictionary error. Locus \(type) not supported")
```
Comment (ystrot) [inline MacawTests/SceneSerialization.swift]: Usually it's easier to just fail there and avoid all these null values handling.

File: Source/model/scene/Text.swift [hunk 5]
```diff
@@ -125,3 +125,13 @@ open class Text: Node {
     }
 
 }
+
+fileprivate func baselineForString(_ string: String) -> Baseline {
```
Comment (ystrot) [inline Source/model/scene/Text.swift]: Remove this method.

File: Source/model/scene/Shape.swift [hunk 6]
```diff
@@ -54,4 +54,5 @@ open class Shape: Node {
 
         return bounds
     }
+
```
Comment (ystrot) [inline Source/model/scene/Shape.swift]: It's better to avoid all these unnecessary changes.

**Comment by ystrot** [comment]
And there is merge conflict now, need to merge with master once again.

File: MacawTests/SceneSerialization.swift (hunk 3 above)
Comment (f3dm76) [reply MacawTests/SceneSerialization.swift]: Then there will be another list of if-s - to add all the factories to the map. Can't see why it would be better

File: MacawTests/SceneSerialization.swift (hunk 4 above)
Comment (f3dm76) [reply MacawTests/SceneSerialization.swift]: null values? These are just unsupported or misspelled node names, or did I not understand you correctly?

File: Source/model/scene/Shape.swift (hunk 6 above)
Comment (f3dm76) [reply Source/model/scene/Shape.swift]: sorry( hard to track them when every file is refactored 5 times

File: MacawTests/SceneSerialization.swift (hunk 4 above)
Comment (ystrot) [reply MacawTests/SceneSerialization.swift]: There is `return null` after this line. If you just throw fatal error you don't need to handle optional value then.

File: Source/model/scene/Shape.swift (hunk 6 above)
Comment (ystrot) [reply Source/model/scene/Shape.swift]: Just make a final review of all your changes before you commit something 😊

Comment (f3dm76) [reply Source/model/scene/Shape.swift]: of course I do. but final review in it client compares with previous commit. First I added a piece of code, and comparing to that commit, I removed a piece of code. Resulting extra line can't be seen at this point. However, I can see it after I commit something right here, which I shall do
